sycl/include/CL/__spirv/spirv_ops.hpp

```diff
@@ -24,120 +24,191 @@
 #ifdef __SYCL_DEVICE_ONLY__
 
 #if (SYCL_EXT_ONEAPI_MATRIX_VERSION > 1)
-#define JOINT_MATRIX_INTEL(T, R, C, L, S, U)                                   \
-  __spv::__spirv_JointMatrixINTEL<T, R, C, L, S, U>
-#else
-#define JOINT_MATRIX_INTEL(T, R, C, L, S, U)                                   \
-  __spv::__spirv_JointMatrixINTEL<T, R, C, L, S>
-#endif // SYCL_EXT_ONEAPI_MATRIX_VERSION
-
-template <typename T, std::size_t R, std::size_t C,
-          __spv::MatrixUse U = __spv::MatrixUse::Unnecessary,
+template <typename T, std::size_t R, std::size_t C, __spv::MatrixUse U,
           __spv::MatrixLayout L = __spv::MatrixLayout::RowMajor,
```

Comment:
we are making the change to dynamic in this PR right?
If yes, the default for layout should be dynamic